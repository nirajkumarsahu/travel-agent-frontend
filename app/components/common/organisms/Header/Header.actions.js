import {
  FETCH_HEADER_DATA,
  HEADER_DATA_SUCCESS,
  HEADER_LOAD_FAILURE
} from "./Header.constants";

export const headerLoadSuccess = data => ({
  type: HEADER_DATA_SUCCESS,
  data
});

export const headerLoadFailure = data => ({
  type: HEADER_LOAD_FAILURE,
  data
});

export const fetchHeaderData = () => ({
  type: FETCH_HEADER_DATA
});
