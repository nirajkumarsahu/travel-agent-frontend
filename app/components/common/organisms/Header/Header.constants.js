export const HEADER_DATA_SUCCESS = "HEADER_DATA_SUCCESS";
export const FETCH_HEADER_DATA = "FETCH_HEADER_DATA ";
export const HEADER_LOAD_FAILURE = "HEADER_LOAD_FAILURE";
