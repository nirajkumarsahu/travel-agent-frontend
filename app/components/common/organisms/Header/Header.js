import React from "react";
import PropTypes from "prop-types";

import styledHOC from "lib/styledHOC";
import HeaderLinks from "components/common/molecules/HeaderLinks";
import UserAccount from "components/common/molecules/UserAccount";
import LoginBoard from "components/common/molecules/LoginBoard";
import Logo from "components/common/molecules/Logo";
import Anchor from "components/common/atoms/Anchor";

import styles from "./Header.style";

const Header = props => {
  const {
    className,
    getSVG,
    updateHomePageState,
    user,
    updateComp,
    helpLineNumber,
    // eslint-disable-next-line react/prop-types
    pageName,
    userProfile,
    signUpHeader
  } = props;
  const Call = getSVG("call");
  let logo = null;
  if (signUpHeader) {
    logo = (
      <div className="flex container">
        <div className="logo-enquiry">
          <Logo getSVG={getSVG} pageName={pageName} />
          <span> Travel Agent Booking Tool </span>
        </div>
        <div className="header-right">
          <ul className="top-list">
            <li className="call">
              <Anchor to={`tel:${helpLineNumber}`}>
                <Call />
                <span className="helpline-number">24x7 Helpline</span>
                <span className="tool-tip">
                  Need Help? Give us a call! <br />
                  0124 434 1560
                </span>
              </Anchor>
            </li>
            <li>
              <Anchor to="https://agents.fabhotels.com">Login</Anchor>
            </li>
          </ul>
        </div>
      </div>
    );
  } else {
    logo = (
      <div className="flex container">
        <Logo getSVG={getSVG} pageName={pageName} />
        <div className="flex header-right-top">
          <HeaderLinks
            getSVG={getSVG}
            helpLineNumber={helpLineNumber}
            pageName={pageName}
          />
          {Object.keys(userProfile || {})[0] || Object.keys(user || {})[0] ? (
            <UserAccount
              userDetails={user}
              pageName={pageName}
              updateComp={updateComp}
              getSVG={getSVG}
              userProfile={userProfile}
            >
              {(
                (userProfile && userProfile.firstName) ||
                (user && user.firstName) ||
                " "
              ).substring(0, 1)}
            </UserAccount>
          ) : (
            <LoginBoard
              loginClick={updateHomePageState}
              getSVG={getSVG}
              pageName={pageName}
            />
          )}
        </div>
      </div>
    );
  }
  return <header className={className}>{logo}</header>;
};

Header.propTypes = {
  className: PropTypes.string.isRequired,
  getSVG: PropTypes.func.isRequired,
  updateHomePageState: PropTypes.func.isRequired,
  user: PropTypes.string.isRequired,
  updateComp: PropTypes.string.isRequired,
  helpLineNumber: PropTypes.string.isRequired,
  userProfile: PropTypes.instanceOf(Object).isRequired,
  signUpHeader: PropTypes.bool
};

Header.defaultProps = {
  signUpHeader: false
};

export { Header as HeaderVanilla };
export default styledHOC(Header, styles);
