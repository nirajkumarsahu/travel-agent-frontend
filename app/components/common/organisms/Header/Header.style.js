import { css } from "styled-components";

export default css`
  background: transparent;
  padding-top: 20px;
  padding-bottom: 20px;
  justify-content: space-between;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  z-index: 9;
  .container {
    justify-content: space-between;
  }
  .header-logo svg {
    width: 100px;
  }
  .helpline-number {
    display: none;
  }
  .logo-enquiry {
    .header-logo {
      vertical-align: middle;
      svg {
        width: 126px;
      }
    }
    span {
      font-size: ${props => props.theme.fs_xsmall}px;
      color: ${props => props.theme.color_White};
      display: block;
    }
  }
  .top-list {
    padding: 0;
    margin: 0;
    display: flex;
  }

  .top-list li {
    list-style: none;
    display: flex;
    align-items: center;
    a {
      color: #fff;
      font-size: ${props => props.theme.fs_secondary}px;
    }
  }

  .tool-tip {
    background: #595a5e;
    padding: 12px 16px;
    border-radius: 5px;
    font-size: 14px;
    text-align: center;
    line-height: 1.5;
    position: absolute;
    right: 0px;
    top: 36px;
    min-width: 193px;
    display: none;

    &:before {
      content: "";
      position: absolute;
      right: 14px;
      top: -10px;
      border-right: 10px solid transparent;
      border-left: 10px solid transparent;
      border-bottom: 10px solid #595a5e;
    }
  }

  .call {
    margin-right: 0px;
    a {
      display: flex;
      align-items: center;
      font-size: ${props => props.theme.fs_secondary}px;
      margin-right: 0;
      padding: 8px;
      position: relative;
      color: #fff;
      &:hover .tool-tip {
        display: block;
      }

      svg {
        width: 16px;
        margin: 0 10px 0 0;
      }

      path {
        fill: ${props => props.theme.color_White};
      }
    }
  }

  @media ${props => props.theme.mediaQuery.medium} {
    padding-top: 25px;
    padding-bottom: 22px;
  .header-logo svg {
    width: 180px;
  }
  .logo-enquiry {
    .header-logo {
      vertical-align: middle;
      svg {
        width: 151px;
        height: 28px;
      }
    }
    span {
      font-size: ${props => props.theme.fs_small}px;
      color: ${props => props.theme.color_White};
      display: inline-block;
    }
  }
  .helpline-number {
    display: block;
  }
  .call {
    margin-right: 30px;
  }
  .tool-tip {
    right: 95px;
  }
`;
