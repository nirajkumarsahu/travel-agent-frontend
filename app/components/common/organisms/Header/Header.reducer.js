import { updateError } from "lib/utils";
import { HEADER_DATA_SUCCESS, HEADER_LOAD_FAILURE } from "./Header.constants";

const updateData = (state, data) => ({
  ...state,
  data
});

export default function(state = {}, action) {
  switch (action.type) {
    case HEADER_DATA_SUCCESS:
      return updateData(state, action.data);
    case HEADER_LOAD_FAILURE:
      return updateError(state, "error Occured in header load");
    default:
      return state;
  }
}
