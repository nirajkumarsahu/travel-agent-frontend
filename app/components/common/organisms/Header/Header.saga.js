import { put, takeLatest, call } from "redux-saga/effects";

import { apiFetch } from "lib/utils";
import { headerLoadSuccess, headerLoadFailure } from "./Header.actions";
import { FETCH_HEADER_DATA } from "./Header.constants";

// eslint-disable-next-line
export function* fetchHeaderData(action) {
  try {
    const res = yield call(apiFetch, {
      url: "https://jsonplaceholder.typicode.com/todos/2",
      method: "GET"
    });
    yield put(headerLoadSuccess(res));
  } catch (err) {
    yield put(headerLoadFailure(err));
  }
}

export default function* headerSaga() {
  try {
    yield takeLatest(FETCH_HEADER_DATA, fetchHeaderData);
  } catch (err) {
    yield put(headerLoadFailure(err));
  }
}
