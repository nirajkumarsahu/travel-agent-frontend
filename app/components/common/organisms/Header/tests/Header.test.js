import React from "react";
import { shallow } from "enzyme";
import { HeaderVanilla } from "../Header";

describe("Header Test Cases", () => {
  const props = {
    className: "testClass"
  };
  let output = "";

  beforeEach(() => {
    output = shallow(<HeaderVanilla {...props} />);
  });

  test("should render correctly", () => {
    expect(output).toMatchSnapshot();
  });
});
