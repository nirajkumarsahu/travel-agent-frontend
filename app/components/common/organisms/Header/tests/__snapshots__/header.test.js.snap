// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test Cases should render correctly 1`] = `
<header
  className="testClass"
>
  <Link
    href="/"
  >
    <a>
      Home
    </a>
  </Link>
  <Link
    href="/about"
  >
    <a>
      About
    </a>
  </Link>
</header>
`;
